src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java

```diff
@@ -14,30 +14,158 @@
+  private static final String LINK_INPUT = "link-input";
+  /* Default value if new lecture inputs are empty. */
+  private static final String DEFAULT_VALUE = "";
+  private static final String REDIRECT_URL = "/lecture-view.html";
+
+  /* Pattern used to create a matcher for a video ID. */
+  private static Pattern videoUrlGeneratedPattern;
+  private static DatastoreService datastore;
+
   @Override
   public void init() throws ServletException {
-    // TODO: Implement LectureServlet.
+    datastore = DatastoreServiceFactory.getDatastoreService();
+    videoUrlGeneratedPattern = Pattern.compile(YOUTUBE_VIDEO_URL_PATTERN);
   }
 
   @Override
+  // TODO: Check if URL is valid.
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.sendError(500, "Not Implemented");
+    String videoUrl = getParameter(request, LINK_INPUT, DEFAULT_VALUE);
+    Optional<Entity> existingEntity = checkUrlInDatabase(videoUrl);
+
+    if (existingEntity.isPresent()) {
+      response.sendRedirect(buildRedirectUrl(existingEntity.get()));
+      return;
+    }
+    Entity lectureEntity = createLectureEntity(request);
+    datastore.put(lectureEntity);
+    response.sendRedirect(buildRedirectUrl(lectureEntity));
   }
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.sendError(500, "Not Implemented");
+    List<Lecture> lectures = getLectures();
+    Gson gson = new Gson();
+    response.setContentType("application/json");
+    response.getWriter().println(gson.toJson(lectures));
+  }
+
+  /**
+   * Returns the Entity in database that has {@code url}, or
+   * {@code Optional.empty()} if one doesn't exist.
+   */
+  private Optional<Entity> checkUrlInDatabase(String url) {
+    Query query = new Query(Lecture.ENTITY_KIND);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+    Iterable<Entity> resultsIterable = results.asIterable();
+
+    for (Entity lecture : resultsIterable) {
+      if (lecture.getProperty(Lecture.PROP_URL).equals(url)) {
+        return Optional.of(lecture);
+      }
+    }
+    return Optional.empty();
+  }
+
+  /** Creates and returns {@code lectureEntity} using parameters found in {@code request}. */
+  private Entity createLectureEntity(HttpServletRequest request) {
+    String lectureName = getParameter(request, NAME_INPUT, DEFAULT_VALUE);
+    String videoUrl = getParameter(request, LINK_INPUT, DEFAULT_VALUE);
+    String videoId = getVideoId(videoUrl);
+
+    Entity lectureEntity = new Entity(Lecture.ENTITY_KIND);
+    lectureEntity.setProperty(Lecture.PROP_NAME, lectureName);
+    lectureEntity.setProperty(Lecture.PROP_URL, videoUrl);
+    lectureEntity.setProperty(Lecture.PROP_VIDEO_ID, videoId);
+    return lectureEntity;
+  }
+
+  /** Returns a list of lectures stored in the database. */
```

Comment:
```suggestion
  /** Returns lectures stored in the database. */
```

Reply:
I would omit this Javadoc. Javadocs are only necessary for public methods and if its not obvious what the method doc. In this case the Javadoc does not add any value.

Reply:
Is there a general guide for JavaDoc, I think I'm missing something